cmd/util/ledger/migrations/storage_v6.go

```diff
@@ -1315,12 +1417,31 @@ func (c *ValueConverter) VisitDeployedContractValue(_ *oldInter.Interpreter, _ o
 
 // Type conversions
 
+var testnetNonFungibleTokenContractAddress = func() common.Address {
+	address, err := hex.DecodeString("631e88ae7f1d7c20")
+	if err != nil {
+		panic(err)
+	}
+
+	return common.BytesToAddress(address)
+}()
+
 func ConvertStaticType(staticType oldInter.StaticType) newInter.StaticType {
 	switch typ := staticType.(type) {
 	case oldInter.CompositeStaticType:
-		return newInter.NewCompositeStaticType(typ.Location, typ.QualifiedIdentifier)
+		location := compositeTypeLocation(typ.Location)
+		return newInter.NewCompositeStaticType(location, typ.QualifiedIdentifier)
 
 	case oldInter.InterfaceStaticType:
+		// NonFungibleToken.NFT is a struct, but is stored as an interface type.
+		// Rectify this by returning a composite static type.
+		if location, ok := typ.Location.(common.AddressLocation); ok {
+			if location.Address == testnetNonFungibleTokenContractAddress &&
+				typ.QualifiedIdentifier == "NonFungibleToken.NFT" {
+				return newInter.NewCompositeStaticType(location, typ.QualifiedIdentifier)
+			}
+		}
```

Comment:
I think this might be a general problem, not just a NFT contract-specific problem.
`NonFungibleToken.NFT` is a "type requirement", a type in an interface which requires the concrete type to provide a concrete type for. I don't remember how this is is implemented in the checker, but I think as an interface type. 

Is it actually a problem that the static type is an interface type here?

Reply:
Yeah, this is a general problem I think. And yeah, the problem is that the static type is stored as an interface type, but in the elaboration, it is in the composite types list.
This could be due to several problems:
-  `NonFungibleToken.NFT` was initially defined as an interface and later changed to a struct/resource?
- Something might not be correct in static-type creation in cadence. (or was not correct, and fixed now?)
- Something went wrong in inferring the static type during V5

I also ran into another very simmilar one, with a static type `Foo.Bar`, where now both `Foo` and `Bar` are concrete types. So I'm starting to feel it could be the first case. But need to investigate a bit more.